src/EditorFeatures/CSharp/DecompiledSource/CSharpDecompiledSourceFormattingRule.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Formatting.Rules;
+
+namespace Microsoft.CodeAnalysis.Editor.CSharp.DecompiledSource
+{
+    internal class CSharpDecompiledSourceFormattingRule : AbstractFormattingRule
+    {
+        public static readonly AbstractFormattingRule Instance = new CSharpDecompiledSourceFormattingRule();
+
+        private CSharpDecompiledSourceFormattingRule()
+        {
+        }
+
+        public override AdjustNewLinesOperation GetAdjustNewLinesOperation(
+            SyntaxToken previousToken, SyntaxToken currentToken, AnalyzerConfigOptions options, in NextGetAdjustNewLinesOperation nextOperation)
+        {
+            var operation = GetAdjustNewLinesOperation(previousToken, currentToken);
+            return operation ?? nextOperation.Invoke();
+        }
+
+        private AdjustNewLinesOperation GetAdjustNewLinesOperation(SyntaxToken previousToken, SyntaxToken currentToken)
+        {
+            // To help code not look too tightly packed, we place a blank line after every statement that ends with a
+            // `}` (unless it's also followed by another `}`).
```

Comment:
What about:

```csharp
do
{
}
while (true)`
```

Reply:
will also be ok. 

but let me look into writing tests that just exercise the formatting portion of this.

Reply:
tests added.